src/main/java/org/cryptomator/common/CatchingExecutors.java

```diff
@@ -0,0 +1,99 @@
+package org.cryptomator.common;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javafx.application.Platform;
+import javafx.concurrent.Task;
+import java.util.Objects;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+//Inspired by: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/ThreadPoolExecutor.html#afterExecute(java.lang.Runnable,java.lang.Throwable)
+public final class CatchingExecutors {
+
+	private static final Logger LOG = LoggerFactory.getLogger(CatchingExecutors.class);
+
+	private CatchingExecutors() { /* NO-OP */ }
+
+	public static class CatchingScheduledThreadPoolExecutor extends ScheduledThreadPoolExecutor {
+
+		public CatchingScheduledThreadPoolExecutor(int corePoolSize, ThreadFactory threadFactory) {
+			super(corePoolSize, threadFactory);
+		}
+
+		@Override
+		protected void afterExecute(Runnable runnable, Throwable throwable) {
+			super.afterExecute(runnable, throwable);
+			afterExecute0(runnable, throwable);
+		}
+	}
+
+	public static class CatchingThreadPoolExecutor extends ThreadPoolExecutor {
+
+		public CatchingThreadPoolExecutor(int corePoolSize, int maximumPoolSize, long keepAliveTime, TimeUnit unit, BlockingQueue<Runnable> workQueue, ThreadFactory threadFactory) {
+			super(corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue, threadFactory);
+		}
+
+		@Override
+		protected void afterExecute(Runnable runnable, Throwable throwable) {
+			super.afterExecute(runnable, throwable);
+			afterExecute0(runnable, throwable);
+		}
+	}
+
+	private static void afterExecute0(Runnable runnable, Throwable throwable) {
+		if (throwable == null) {
+			if (runnable instanceof Task<?>) {
+				afterExecuteTask(Thread.currentThread(), (Task<?>) runnable);
+				return;
+			}
+			throwable = getThrowable(runnable);
+		}
+
+		if (throwable != null) {
+			callHandler(Thread.currentThread(), throwable);
+		}
```

Comment:
```suggestion
		if (throwable != null) {
			callHandler(Thread.currentThread(), throwable);
		} else if (runnable instanceof Task<?> t) {
			afterExecuteTask(Thread.currentThread(), t);
		} else if (runnable instanceof Future<?> f) {
			afterExecuteFuture(f);
		}
```

Reply:
fair enough, nonetheless the code can be structured. Just reorder the ifs.

Reply:
nevermind, just saw that you reordered this already, giving "throwable" the highest priority